lua/go/install.lua

```diff
@@ -1,53 +1,42 @@
-local uv = vim.loop
-local gopath = vim.fn.expand("$GOPATH")
-local gobinpath = gopath .. "/bin/"
 local url = {
-  golines = "segmentio/golines",
-  gofumpt = "mvdan/gofumpt",
-  gofumports = "mvdan/gofumpt",
-  gomodifytags = "fatih/gomodifytags",
-  gotsts = "cweill/gotests",
-  iferr = 'koron/iferr',
-  fillstruct = 'davidrjenni/reftools/cmd/fillstruct',
-  fixplurals = 'davidrjenni/reftools/cmd/fixplurals',
-  fillswitch = 'davidrjenni/reftools/cmd/fillswitch',
+  gofumpt      = "mvdan.cc/gofumpt",
+  gofumports   = "mvdan.cc/gofumpt",
+  golines      = "github.com/segmentio/golines",
+  gomodifytags = "github.com/fatih/gomodifytags",
+  gotsts       = "github.com/cweill/gotests",
+  iferr        = 'github.com/koron/iferr',
+  fillstruct   = 'github.com/davidrjenni/reftools/cmd/fillstruct',
+  fixplurals   = 'github.com/davidrjenni/reftools/cmd/fixplurals',
+  fillswitch   = 'github.com/davidrjenni/reftools/cmd/fillswitch',
 }
 
 local function install(bin)
-  local state = uv.fs_stat(gobinpath .. bin)
-  if not state then
+  local state = vim.cmd("!which " .. bin)
```

Comment:
This is the only part that doesn't really work yet. Most of what I've tried so far was a bit awkward

Reply:
Is there a reason for using "!witch" instead of fs_state?

Reply:
another concern I have is for none linux/mac system, the `which` command may not exist.  It may use `where` instead. and the command output may also not same.

I think the concern you have here is the go-binary is not installed in 'gobinpath'. Is it possible to search multiple directories using fs_state.

Reply:
My reasoning for using `which` was that it is a more dynamic approach. As every system might be a bit different, and have binaries located in different placed. So instead of hardcoding them you give the flexibility to the user as long as they are in the `$PATH` variable. Otherwise you'd have to hardcode a bunch of paths, and some edge case might still occur. Therefore I thought a `which`-like approach would be most flexible.

You're right on the windows compatibility, hadn't thought of that yet